vinage.js

```diff
@@ -255,7 +249,9 @@ var vinage = (function(){
 			return pseudoClone;
 		}
 		function mod(dividend, divisor) {
-			return (dividend%divisor + divisor) % divisor;
+			if (!isFinite(divisor)) return divdend;
+			while (dividend < 0 || dividend >= divisor) dividend = (dividend + divisor) % divisor;
```

Comment:
This is going to be very inefficient. A better way to achieve this without having to loop is `(dividend%divisor + divisor) % divisor`.

Example:

> -20 % 3 = -2

The result will always be less than the divisor, it will need one increment of a divisor to be the real modulo.

So you can do:

> -20 % 3 + 3

which is 1, so that works fine.

However, if your divisor is greater than 0, this formula will fuck up:

> 20 % 3 + 3 = 5

But if you apply the modulo of on this **again**, it works fine both when the divisor is > 0 and when it's < 0.

> (20 % 3 + 3)%3 = 2
> (-20 % 3 + 3)%3 = 1

Reply:
Another approach which is maybe more efficient (I don't like the double modulo method) is to take the absolute of the value (-20 gets 20) and do the modulo once. For 20 you'll get `20 % 3 = 3` for minus 20 you'll need to substract the modulo from the divisor, so that `abs(-20) % 3 = 2` and `3 - 2 = 1`. Which will have less modulo, though I don't know if it's somehow more efficient.

Reply:
Oops sorry I did not respond in the right place https://github.com/KordonBleu/vinage/pull/2#issuecomment-349318669.